common/patches/0001-Build-only-what-we-use-to-test-EBBR-on-32b-arm.patch

```diff
@@ -0,0 +1,1587 @@
+ dt_sources = \
+@@ -47,24 +41,13 @@ endif
+ #  Components of the fwts core helper library libfwts
+ #
+ libfwts_la_SOURCES = 		\
+-	fwts_ac_adapter.c 	\
+-	fwts_acpi_object_eval.c \
+-	fwts_acpi_tables.c 	\
+-	fwts_acpi.c 		\
+-	fwts_acpid.c 		\
+ 	fwts_alloc.c 		\
+ 	fwts_arch.c 		\
+ 	fwts_args.c 		\
+ 	fwts_backtrace.c	\
+-	fwts_battery.c 		\
+ 	fwts_binpaths.c 	\
+-	fwts_button.c 		\
+ 	fwts_checkeuid.c 	\
+ 	fwts_checksum.c 	\
+-	fwts_clog.c		\
+-	fwts_cmos.c 		\
+-	fwts_coreboot.c		\
+-	fwts_coreboot_cbmem.c	\
+ 	fwts_cpu.c 		\
+ 	fwts_dump.c 		\
+ 	fwts_dump_data.c 	\
+@@ -77,13 +60,9 @@ libfwts_la_SOURCES = 		\
+ 	fwts_get.c 		\
+ 	fwts_gpe.c 		\
+ 	fwts_guid.c 		\
+-	fwts_hwinfo.c 		\
+-	fwts_iasl.c 		\
+ 	fwts_interactive.c 	\
+ 	fwts_ioport.c		\
+-	fwts_ipmi.c		\
+ 	fwts_json.c		\
+-	fwts_kernel.c 		\
+ 	fwts_keymap.c 		\
+ 	fwts_klog.c 		\
+ 	fwts_olog.c		\
+@@ -99,22 +78,16 @@ libfwts_la_SOURCES = 		\
+ 	fwts_modprobe.c		\
+ 	fwts_multiproc.c 	\
+ 	fwts_oops.c 		\
+-	fwts_pci.c		\
+ 	fwts_pipeio.c 		\
+ 	fwts_release.c		\
+ 	fwts_scan_efi_systab.c 	\
+ 	fwts_set.c 		\
+-	fwts_smbios.c 		\
+ 	fwts_stringextras.c 	\
+ 	fwts_summary.c 		\
+ 	fwts_text_list.c 	\
+-	fwts_tpm.c		\
+ 	fwts_tty.c 		\
+ 	fwts_uefi.c 		\
+-	fwts_wakealarm.c 	\
+-	fwts_pm_method.c	\
+ 	fwts_safe_mem.c		\
+-	fwts_pm_debug.c		\
+ 	$(dt_sources)
+ 
+ -include $(top_srcdir)/git.mk
+diff --git a/src/lib/src/fwts_fileio.c b/src/lib/src/fwts_fileio.c
+index 83a62bc4..eac7fdec 100644
+--- a/src/lib/src/fwts_fileio.c
++++ b/src/lib/src/fwts_fileio.c
+@@ -62,41 +62,3 @@ fwts_list* fwts_file_open_and_read(const char *file)
+ 
+ 	return list;
+ }
+-
+-/*
+- *  fwts_gzfile_read()
+- *	read given gz file and return contents as a list of lines
+- */
+-fwts_list *fwts_gzfile_read(gzFile *fp)
+-{
+-	fwts_list *list;
+-	char buffer[8192];
+-
+-	if ((list = fwts_list_new()) == NULL)
+-		return NULL;
+-
+-	while (gzgets(*fp, buffer, sizeof(buffer)) != NULL) {
+-		buffer[strlen(buffer) - 1] = '\0';	/* Chop off "\n" */
+-		fwts_text_list_append(list, buffer);
+-	}
+-
+-	return list;
+-}
+-
+-/*
+- *  fwts_gzfile_open_and_read()
+- *	open and read gz file and return contents as a list of lines
+- */
+-fwts_list* fwts_gzfile_open_and_read(const char *file)
+-{
+-	gzFile fp;
+-	fwts_list *list;
+-
+-	if ((fp = gzopen(file, "r")) == Z_NULL)
+-		return NULL;
+-
+-	list = fwts_gzfile_read(&fp);
+-	(void)gzclose(fp);
+-
+-	return list;
+-}
+diff --git a/src/lib/src/fwts_framework.c b/src/lib/src/fwts_framework.c
+index 725a7e0b..1ed17432 100644
+--- a/src/lib/src/fwts_framework.c
++++ b/src/lib/src/fwts_framework.c
+@@ -31,7 +31,6 @@
+ #include <sys/time.h>
+ 
+ #include "fwts.h"
+-#include "fwts_pm_method.h"
+ 
+ 
+ #define ACS_VERSION "v1.0"
+diff --git a/src/lib/src/fwts_uefi.c b/src/lib/src/fwts_uefi.c
+index 2dd1727e..61aa7b9d 100644
+--- a/src/lib/src/fwts_uefi.c
++++ b/src/lib/src/fwts_uefi.c
+@@ -41,7 +41,7 @@ typedef struct {
+ 	uint8_t		guid[16];
+ 	uint64_t	datalen;
```

Comment:
Query: Why the datatype of other 64 bit members not been modified to 32 bit?

Reply:
Hi @edhay,
The `UEFI_STATUS` type is defined by the [UEFI spec.](https://uefi.org/specs/UEFI/2.10/02_Overview.html#data-types) as a `UINTN`, which is 4 bytes on 32b processors and 8 bytes on 64b processors. As FWTS did hardcode the status variables to `uint64_t`, I had to adapt the code to be either 32b or 64b depending on the processor, hence the `efi_status_t`.
AFAICT, other datatypes do not have the problem. (To be fair I see in the spec that `EFI_TPL` is also defined as `UINTN`, which means that other places in FWTS might need to be adapted.)